draft-ietf-quic-ack-frequency.md

```diff
@@ -411,18 +411,25 @@ controllers.
 ## Congestion Control
 
 A sender needs to be responsive to notifications of congestion, such as
-a packet loss or an ECN CE marking. Also, window-based congestion controllers
-that strictly adhere to packet conservation, such as the one defined in
-{{QUIC-RECOVERY}}, rely on receipt of acknowledgments to send additional data into
-the network, and will suffer degraded performance if acknowledgments are delayed
-excessively.
-
-To enable a sender to respond to potential network congestion, a sender SHOULD
-cause a receiver to send an acknowledgement at least once per RTT if there are
-unacknowledged ack-eliciting packets in flight. A sender can accomplish this by
-sending an IMMEDIATE_ACK frame once per round-trip time (RTT), or it can set the
-Ack-Eliciting Threshold and Request Max Ack Delay values to be no more than a
-congestion window and an estimated RTT, respectively.
+a packet loss or an ECN CE marking.  To enable a sender to respond to potential
+network congestion, a sender SHOULD cause a receiver to send an acknowledgement
```

Comment:
Why is it the sender's responsibility to ensure that the receiver sends acknowledgments?  I have a fair idea, but it needs a bit of explanation as it is not obvious.

My first thought was that maybe the receiver shouldn't accept a delay that exceeded the RTT.  That might still be a good safety measure, even if the view of RTT is subjective and it might disagree with the sender on how that limit applies.

Reply:
There are cases where a delay larger than RTT, at least min RTT could be appropriate.  One example is when RTT << timer granularity, ie: 10 us network latency and 1ms timers in a datacenter.  Now that I think about it, I will call that case out.

Reply:
I would expect you might do things differently in a DC -  we could declare a minimum here...

Reply:
Yes, I'm also not sure about this case. Generally I think you should try to at least send one ACK per RTT. If you timer granularity is lower than RTT, you could still use a packet threshold to achieve this.

If you want to diverge from this guidance in a DC is something we don't really control. But I think at least once per RTT is the right guidance for the Internet.

Reply:
I'm not sure why you suggest /one ACK per RTT/ ? In the measurements I have seen, several (e.g. 4) per RTT are beneficial to be avoid the effects of path variation and loss on the return path. But, agree on the backup of using the packet threshold to achieve this, which seems also to work well if the RTT varies.